plugins/auto-sizes/tests/test-improve-calculate-sizes.php

```diff
@@ -1259,6 +1272,134 @@ public function data_image_block_with_parent_columns_and_its_parent_group_block(
 		);
 	}
 
+	/**
+	 * Verifies that the post featured image block does not render when no featured image is set for the post.
+	 */
+	public function test_post_featured_image_block_without_featured_image(): void {
+		$block_content = '<!-- wp:post-featured-image /-->';
+
+		// Set up global $post so 'the_content' filter works as expected.
+		global $post;
+		$post = get_post( self::$post_id );
+		setup_postdata( $post );
+
+		$result = apply_filters( 'the_content', $block_content );
+
+		$this->assertStringContainsString( '', $result );
+
+		wp_reset_postdata();
+	}
+
+	/**
+	 * Test that the post featured image block renders correctly with different image sizes.
+	 *
+	 * @dataProvider data_post_featured_image_block_image_sizes
+	 *
+	 * @param string $image_size Image size.
+	 * @param string $expected   Expected output.
+	 */
+	public function test_post_featured_image_block_with_different_image_size( string $image_size, string $expected ): void {
+		update_post_meta( self::$post_id, '_thumbnail_id', self::$image_id );
+
+		$block_content = '<!-- wp:post-featured-image {"sizeSlug":"' . $image_size . '"} /-->';
+
+		// Set up global $post so 'the_content' filter works as expected.
+		global $post;
+		$post = get_post( self::$post_id );
+		setup_postdata( $post );
+
+		$result = apply_filters( 'the_content', $block_content );
+
+		// Check that the featured image block renders the image and has a sizes attribute.
+		$this->assertStringContainsString( 'wp-block-post-featured-image', $result );
+		$this->assertStringContainsString( $expected, $result );
+
+		wp_reset_postdata();
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array<array<string>> The image sizes.
+	 */
+	public function data_post_featured_image_block_image_sizes(): array {
```

Comment:
```suggestion
	public static function data_post_featured_image_block_image_sizes(): array {
```

Reply:
Updated in https://github.com/WordPress/performance/pull/2084/commits/5e976a882eb73e896e71d2dae4bb23940ead86b9